internal/server/v2/observation/contained_in.go

```diff
@@ -190,27 +201,34 @@ func FetchContainedIn(
 					sort.Sort(ranking.CohortByRank(cohorts))
 					for _, cohort := range cohorts {
 						facet := util.GetFacet(cohort)
-						if filter != nil && filter.Domains != nil {
-							url, err := url.Parse(facet.ProvenanceUrl)
-							if err != nil {
-								return nil, err
-							}
-							matchedDomain := false
-							for _, domain := range filter.Domains {
-								// To match domain or subdomain. For example, a provenance url of
-								// abc.xyz.com can match filter "xyz.com" and "abc.xyz.com".
-								if strings.HasSuffix(url.Hostname(), domain) {
-									matchedDomain = true
+						facetID := util.GetFacetID(facet)
+						// If there are facet filters, check that the cohort matches at
+						// least one filter. Otherwise, skip.
+						if facetFilters, ok := variableFacetFilters[variable]; ok {
```

Comment:
maybe make this a function that like: checkCohort(facet, filters[]) => bool? and this can be reused in the other file as well.

Reply:
done